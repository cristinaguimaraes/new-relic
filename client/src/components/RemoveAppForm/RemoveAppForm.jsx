import React, {useState} from 'react';
import {Helper, Input, Label, Form, Button, ButtonWrapper} from './styled';
import { func} from "prop-types";
import {Title} from "../AddAppForm/styled";

const RemoveAppForm  =({onDeleteApp, onShowToastMessage}) => {
  const [name, setName] = useState();
  const [error, setError] = useState();

  const handleOnSubmit = (event) => {
    event.preventDefault();

    if (!name) {
        setError('Please add the name');
    } else {

    onDeleteApp({name});
    onShowToastMessage({messageKey: 'App successfully deleted'});
    setName('');
  }
};

    const handleOnChange = (event) => {
        setName(event.target.value);
        setError( '')
    };

    return (
        <Form onSubmit={handleOnSubmit} >
            <Title>Remove App</Title>
          <Label>App Name:</Label>
          <Input name='name' type='text' value={name} onChange={handleOnChange}/>
          {error && <Helper>{error}</Helper>}
          <ButtonWrapper>
              <Button  type="submit" >Remove</Button>
          </ButtonWrapper>
        </Form>
    )

};

RemoveAppForm.propTypes = {
  onDeleteApp: func.isRequired,
  onShowToastMessage: func.isRequired
};

export { RemoveAppForm};
