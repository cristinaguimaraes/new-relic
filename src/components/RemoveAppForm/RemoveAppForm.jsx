import React, {useState} from "react";
import {
  Helper,
  Input,
  Label,
  Form,
  Button,
  ButtonWrapper,
  Title
} from "./styled";
import {func, string, shape, arrayOf} from "prop-types";

const RemoveAppForm = ({onDeleteApp, onShowToastMessage, data}) => {
  const [name, setName] = useState('');
  const [error, setError] = useState();

  const handleOnSubmit = event => {
    event.preventDefault();

    if (!name) {
      setError("Please add the name");
    } else {
      const appIndex = data.findIndex(eachApp => eachApp.name === name);
      if (appIndex !== -1) {
        onDeleteApp({appIndex});
        //If 'onDeleteApp' was dispatching an action calling the API, the toast should appear just in case of Success. It is here because of the scope of the project.
        onShowToastMessage({messageKey: "App successfully deleted", type: 'success'});
      } else {
        onShowToastMessage({messageKey: "App's name doesn't exist", type: 'failure'});
      }
      setName("");
    }
  };

  const handleOnChange = event => {
    setName(event.target.value);
    setError("");
  };

  return (
    <Form onSubmit={handleOnSubmit}>
      <Title>Remove App</Title>
      <Label>App Name:</Label>
      <Input name="name" type="text" value={name} onChange={handleOnChange}/>
      {error && <Helper>{error}</Helper>}
      <ButtonWrapper>
        <Button type="submit">Remove</Button>
      </ButtonWrapper>
    </Form>
  );
};

RemoveAppForm.propTypes = {
  onDeleteApp: func.isRequired,
  onShowToastMessage: func.isRequired,
  data: arrayOf(shape({name: string}))
};

export {RemoveAppForm};
