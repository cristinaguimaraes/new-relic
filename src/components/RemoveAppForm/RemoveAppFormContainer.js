import { RemoveAppForm } from "./RemoveAppForm";
import { onDeleteApp, onShowToastMessage } from "../../redux/actions/creators";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";

const mapStateToProps = state => ({
  data: state.apps.data
});

const mapDispatchToProps = {
  onDeleteApp,
  onShowToastMessage
};

export const RemoveAppFormContainer = withRouter(
  connect(mapStateToProps, mapDispatchToProps)(RemoveAppForm)
);
