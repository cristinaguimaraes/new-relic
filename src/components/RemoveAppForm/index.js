import { RemoveAppFormContainer } from "./RemoveAppFormContainer";
export { RemoveAppFormContainer as RemoveAppForm };
