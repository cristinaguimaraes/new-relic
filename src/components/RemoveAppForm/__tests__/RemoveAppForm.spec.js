import {RemoveAppForm} from "../RemoveAppForm";
import {shallow} from "enzyme";
import React from "react";

describe("RemoveAppForm component", () => {
  test("should render properly", () => {
    expect(
      shallow(
        <RemoveAppForm onDeleteApp={jest.fn()} onShowToastMessage={jest.fn()}
                       data={[{name: 'Awesome Wooden Sausages - Schaefer - Hegmann Inc'}]}>
          Test
        </RemoveAppForm>
      )
    ).toMatchSnapshot();
  });
});
