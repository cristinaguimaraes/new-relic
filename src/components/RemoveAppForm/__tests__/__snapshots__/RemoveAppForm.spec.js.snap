// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveAppForm component should render properly 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.div>
    Remove App
  </styled.div>
  <styled.label>
    App Name:
  </styled.label>
  <styled.input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <styled.div>
    <styled.button
      type="submit"
    >
      Remove
    </styled.button>
  </styled.div>
</styled.form>
`;
